projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -416,9 +421,66 @@ internal void Write(ReadOnlyMemory<byte> memory)
 
         internal ValueTask WriteAsync(ReadOnlyMemory<byte> memory)
         {
+            SetNetworkTags(Activity.Current);
             return _frameHandler.WriteAsync(memory);
         }
 
+        private void SetNetworkTags(Activity? activity)
+        {
+            if (activity != null && activity.IsAllDataRequested)
+            {
+                activity.SetTag("network.transport", "tcp");
+                switch (_frameHandler.RemoteEndPoint.AddressFamily)
+                {
+                    case AddressFamily.InterNetworkV6:
+                        activity.SetTag("network.type", "ipv6");
+                        break;
+                    case AddressFamily.InterNetwork:
+                        activity.SetTag("network.type", "ipv4");
+                        break;
+                }
+
+                if (!string.IsNullOrEmpty(_frameHandler.Endpoint.HostName))
+                {
+                    activity
+                        .SetTag("server.address", _frameHandler.Endpoint.HostName)
+                        .SetTag("destination.address", _frameHandler.Endpoint.HostName);
```

Comment:
no need to set `destination.address` and `destination.port` - it's intended for things like non-directional low-level networking and does not really apply to messaging.